crates/fs/traits.rs

```diff
@@ -0,0 +1,66 @@
+use crate::BlockStore;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use libipld::{error::SerdeError, serde as ipld_serde, Ipld};
+use serde::{Serialize, Serializer};
+
+//--------------------------------------------------------------------------------------------------
+// Traits
+//--------------------------------------------------------------------------------------------------
+
+/// Implements getting a unique identifier for a node.
+pub trait Id {
+    /// Gets an identifier for the node.
+    fn get_id(&self) -> String;
+}
+
+/// Implements deep equality check for two types.
+#[async_trait(?Send)]
+pub trait DeepEq {
```

Comment:
Technically `Eq` is already supposed to be a sort of deep equality (I think).
The biggest difference is that this is deep equality for IPLD. So maybe we call this `IpldEq`? Or `LinkedDataEq` or something like that? To a reader that may explain better why you need to provide it with a blockstore.